### PR TITLE
2831 define app pages order when displayed list of pages

### LINKED ISSUE #2831: Define app pages order when displayed list of pages
## What Feature Can Be Enhanced?

Define some order in which pages will be displayed. Random order looks strange.

Maybe we can use system pages on top, and then regular pages sorted alphabetically under the system ones?

Currently it looks like this
Screenshot 2023-07-22 at 12 27 51

The suggested list for this case could be
_app
404
500
About
Blog
Guestbook
Home

@webberwang what do you think? Does this make sense?

### PR DESCRIPTION
**Pull request by CreativeDev0809**
## Description

Fixed the sort pages of apps with this order: _app, normal pages, 404, 500

## Related Issue(s)

Fixes #2831

### CODE REVIEW

File: libs/frontend/domain/app/src/store/app.service.ts [hunk 1]
```diff
@@ -325,10 +326,33 @@ export class AppService
...
       return undefined
     }
 
+    /**
+     * Sort pages for app. Order is app, custom pages, 404, 500
+     */
+
+    const _pages: Array<BuilderPageFragment> = appData.pages
+    _pages.sort((a, b) => {
+      if (a.name === '_app') {
```
Comment (webberwang) [inline libs/frontend/domain/app/src/store/app.service.ts:341]: we should use enum here

File: libs/frontend/domain/page/src/use-cases/get-pages/PageList.tsx [hunk 2]
```diff
@@ -9,6 +9,8 @@ interface PageListProps {
 }
 
 export const PageList = observer(({ app }: PageListProps) => {
+  console.log(app.pages)
```
Comment (webberwang) [inline libs/frontend/domain/page/src/use-cases/get-pages/PageList.tsx:12]: remove the log

**Review by webberwang — CHANGES_REQUESTED** [review]

File: libs/frontend/domain/app/src/store/app.service.ts [hunk 3]
```diff
@@ -325,10 +326,39 @@ export class AppService
...
+
     if (!appData) {
       return undefined
     }
 
+    /**
+     * Sort pages for app. Order is app, custom pages, 404, 500
+     */
+
+    enum PageType {
```
Comment (webberwang) [inline libs/frontend/domain/app/src/store/app.service.ts:339]: the enum already exists in our project, it's inside shared abstract folder I believe

**Review by webberwang — CHANGES_REQUESTED** [review]

File: libs/frontend/domain/app/src/store/app.service.ts (hunk 3 above)
Comment (CreativeDev0809) [reply libs/frontend/domain/app/src/store/app.service.ts]: Fixed using compare with page kind(used enum IPageKind in abstract/core). Please check it.

**Pull request merged**